### ISSUE TITLE
[HOLD for payment 2023-03-24] [$1000] Search box does not lose focus on scroll on manage member page 

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15695]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app
2. Open settings
3. Open Workspaces
4. Open any workspace which has many members
5. Open manage members
6. Tap on search and scroll through members

## Expected Result:
Search box should lose focus as it does throughout the app

## Actual Result:
Search box does not lose focus

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.79-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/223282938-951d1e5a-c519-47ec-a373-d71f2101d8b1.mp4

https://user-images.githubusercontent.com/43996225/223282960-96f4353d-f6af-4629-b017-9e85c6f74f05.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1678095039391749

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01840c148c49daa5a2</li>
        <li>Upwork Job ID: 1633135246951043072</li>
        <li>Last Price Increase: 2023-03-07</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @isabelastisser (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by isabelastisser** [comment]
I'm able to reproduce this!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01840c148c49daa5a2

**Comment by MelvinBot** [comment]
Current assignee @isabelastisser is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @sobitneupane (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @flodnv (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by logiclingusitic** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Search box does not lose focus on scroll on manage member page #15695
### What is the root cause of that problem?
On manage Member page, when we click on search for a member and scroll down, the search box does not lose focus because the search box is not a part of the scrollable area. Also the first purpose of the search button is to highlight the search box.

### What changes do you think we should make in order to solve the problem?
To make a search box lose focus when a user scrolls we can use the `onBlur` event handler and the `useState` hook. 
We first create a state variable `isSearchFocused` using the `useState` hook and initialize it to `false`. Then, we define the `handleSearchBlur` function that will be called when the search box loses focus. This function simply sets the `isSearchFocused` state to `false`.

We also define the `handleScroll` function that will be called when the user scrolls within the containing `<div>` element. This function checks if the search box is currently focused (`isSearchFocused` is `true`) and if so, it sets `isSearchFocused` to `false`.

**Comment by wildan-m** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The search box does not lose focus on scrolling on the manage member page

### What is the root cause of that problem?

The focus and keyboard is not dismissed during scrolling

### What changes do you think we should make in order to solve the problem?

Adding `Keyboard.dismiss();` will resolve the issue.

https://github.com/Expensify/App/blob/7e8e8a1d881914abe7c13768eb13e4930cb8546b/src/pages/workspace/WorkspaceMembersPage.js#L401-L409

```
                                        onScroll={() => {
                                            Keyboard.dismiss();
                                        }}
```

### What alternative solutions did you explore? (Optional)
N/A

**Comment by MelvinBot** [comment]
📣 @wildan-m! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by wildan-m** [comment]
Contributor details
Your Expensify account email: wildevemail@gmail.com
Upwork Profile Link: https://www.upwork.com/fl/~018b6936cd5349fa66

**Comment by MelvinBot** [comment]
⚠️ Missing/invalid email or upwork profile link. Please make sure you add both your Expensify email and Upwork profile link in the format specified.

**Comment by wildan-m** [comment]
Contributor details
Your Expensify account email: wildevemail@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/wildanmuhlis

**Comment by MelvinBot** [comment]
⚠️ Missing/invalid email or upwork profile link. Please make sure you add both your Expensify email and Upwork profile link in the format specified.

**Comment by flodnv** [comment]
@sobitneupane what are your thoughts on these proposals?

**Comment by s77rt** [comment]
I don't see how this is a bug. Personally I think the expected behaviour is the buggy version and what we have now is better.

**Comment by dhanashree-sawant** [comment]
Hi @s77rt, In the bug report of slack, I have raised it as inconsistency issue. Issue is not that we are using better version but issue is that not all the places have same version.

**Comment by sobitneupane** [comment]
> I don't see how this is a bug. Personally I think the expected behaviour is the buggy version and what we have now is better.

@s77rt Can you please explain a little more? You have mentioned that the expected behavior is buggy. 

Is there any previous discussions on this?

**Comment by sobitneupane** [comment]
@Prince-Mendiratta 

> We can copy over the logic from OptionsList to dismiss the keyboard when the list is scrolled.

Why don't we use `OptionsList` instead of `FlatList` rather than copying the logic? Moreover, is it possible to use `OptionsSelector` in `WorkspaceMembersPage`

> method to dismiss keyboard on website (mWeb) and native devices is different

Can you please explain how and why is it different? [This proposal ](https://github.com/Expensify/App/issues/15695#issuecomment-1459883124) looks simple and does the job pretty well.

**Comment by situchan** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Search box does not lose focus and hide keyboard while scrolling on manage members page

### What is the root cause of that problem?
This is default behavior of RN ScrollView, FlatList. While scrolling list, keyboard is not dismissed automatically unless we insert custom logic.

### What changes do you think we should make in order to solve the problem?
We can dismiss keyboard on scroll event. But only when user scrolls the screen.
`onScroll` event is triggered when data changes so we should handle it as well. Otherwise it will cause regression of #11801
So the solution is to apply same approach in WorkspaceMembers page as what we already did in OptionsList in [this PR](https://github.com/Expensify/App/pull/12057).
https://github.com/Expensify/App/blob/57a2129b9288cbbee339514822a5dd64e73b6804/src/components/OptionsList/index.js#L51-L57

**Comment by s77rt** [comment]
I think the title, platforms and videos are not correct and misleading, this is not about focus itself but rather on keyboard being dismissed and I'm pretty sure this has nothing to do with Web. In my [comment](https://github.com/Expensify/App/issues/15695#issuecomment-1465294381) above I was referring to Web.

For mWeb/Native, looks like the keyboard on OptionList is intended to be dismissed as this behaviour was introduced by this [PR](https://github.com/Expensify/App/pull/5753) to fix https://github.com/Expensify/App/issues/4994. However FlatList does not have this behaviour per default.

___

@sobitneupane Regarding @Prince-Mendiratta's comment

> method to dismiss keyboard on website (mWeb) and native devices is different

- For Native: we use `onScrollBeginDrag`
- For mWeb: no explicit answer, but currently we use a conditional `onScroll`  

___

Sorry for the unnecessary comments @sobitneupane. Leaving this to you now :+1:

**Comment by dhanashree-sawant** [comment]
Hi @s77rt , I agree it has nothing to do with web, issue is only on mobile app and mWeb, sorry for including them in platforms, that is mistake from my end.

**Comment by sobitneupane** [comment]
Thanks for proposal everyone.

[Proposal](https://github.com/Expensify/App/issues/15695#issuecomment-1465742470) from @situchan looks good to me.


🎀👀🎀 C+ reviewed
@flodnv

**Comment by flodnv** [comment]
Thank you for the detailed proposal @situchan, go ahead and submit a PR 🙇

**Comment by MelvinBot** [comment]
📣 @situchan You have been assigned to this job by @flodnv!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by situchan** [comment]
@sobitneupane @flodnv [PR](https://github.com/Expensify/App/pull/15903) is ready for review

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.86-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15903


If no regressions arise, payment will be issued on 2023-03-24. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@sobitneupane / @flodnv] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@sobitneupane / @flodnv] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@sobitneupane / @flodnv] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@isabelastisser] Determine if we should create a regression test for the bug.
- [ ] [@sobitneupane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to the [appropriate location](https://github.com/Expensify/App/blob/main/contributingGuides/TestRail_Navigation_C%2B.md) to ensure the same bug will not reach production again.
- [ ] [@isabelastisser] Review the proposed regression test steps and location.
- [ ] [@isabelastisser] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by dhanashree-sawant** [comment]
Hi @flodnv , @sobitneupane , @isabelastisser, the issue was hold for payment 24th March, can you have a look once you are available?

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
>  The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/15123 The issue was discussed in the PR and considered out of scope.

>  The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/15123#issuecomment-1486815473

> * [ ]  [@sobitneupane / @flodnv] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

https://expensify.slack.com/archives/C049HHMV9SM/p1680008031384569

**Comment by sobitneupane** [comment]
**Regression Test Proposal**

- Login with any account
- Open Manage members page of a workspace
- Tap on search and scroll through members
- Verify that keyboard is dismissed automatically
- Tap on search and type any keyword
- Verify that keyboard is not dismissed while typing

**Comment by isabelastisser** [comment]
- situchanSitu Chandra Shil proposal approved
- sobitneupaneSobit Neupane C+ review 
- Issue reported by: @dhanashree-sawant

**Comment by isabelastisser** [comment]
@situchan , @sobitneupane and @dhanashree-sawant I sent you an offer and rehired you in Upwork now, please accept it and I will process the payment asap. Thanks!

**Comment by isabelastisser** [comment]
All contributors were paid in Upwork and the contract ended!
